modules/planning/traffic_rules/traffic_light.h

```diff
@@ -0,0 +1,49 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file
+ **/
+
+#pragma once
+
+#include "modules/planning/traffic_rules/traffic_rule.h"
+
+namespace apollo {
+namespace planning {
+
+class TrafficLight : public TrafficRule {
+ public:
+  explicit TrafficLight(const TrafficRuleConfig& config);
+
+  virtual ~TrafficLight() = default;
+
+  common::Status ApplyRule(Frame* const frame,
+                           ReferenceLineInfo* const reference_line_info);
+
+ private:
+  void MakeDecisions(Frame* const frame,
+                     ReferenceLineInfo* const reference_line_info);
+  int BuildStopDecision(Frame* const frame,
+                        ReferenceLineInfo* const reference_line_info,
+                        const hdmap::PathOverlap& traffic_light_overlap);
+
+ private:
+  static constexpr char const* const TRAFFIC_LIGHT_VO_ID_PREFIX = "TL_";
```

Comment:
If it's only used in this component, simply put in the `.cc` file and close to where it's used. Header file is some kind of `interface` to external components while this information is unused to them. So I would prefer minimum scope.

Reply:
This variable used to be a gflag to be "configurable". But we haven't really touched it ever since so we moved it inside the class, to make it less flexible.
But still, it is still not really a "local" variable, or should be kept unseen. So I think it's more appropriate to put it in the header file. So we can still be aware of it just in case if we want to change it. 
it's a private member.